File: Problems/ArrayAndStringType/Count_Elt_Greater_CurrAvg/count_elt_greater_curr_avg.py
def countResponseTimeRegressions(responseTimes):
    if len(responseTimes) <= 1: return 0
    
    count:int = 1
    total:int = responseTimes[0]
    to_ret:int = 0
    
    for i in range(1, len(responseTimes)):
        curr:int = responseTimes[i]
        total += curr
        count += 1
        curr_avg = total / count
        if curr > curr_avg:
            to_ret += 1
    
    return to_ret

if __name__ == "__main__":
    # Example test cases
    tests = [
        [100, 120, 80, 140, 130],
        [10, 20, 30, 40],
        [50, 40, 30, 20],
        [5],
        []
    ]

    for idx, arr in enumerate(tests):
        result = countResponseTimeRegressions(arr)
        print(f"Test Case {idx+1}")
        print(f"Input:  {arr}")
        print(f"Output: {result}")
        print("-" * 30)

File: Problems/ArrayAndStringType/Count_Elt_Greater_CurrAvg/countEltGreaterCurrAvg.java
package ArrayAndStringType.Count_Elt_Greater_CurrAvg;

import java.util.*;

public class countEltGreaterCurrAvg {
    
}

class Result {

    /*
     * Complete the 'countResponseTimeRegressions' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY responseTimes as parameter.
     */

    public static int countResponseTimeRegressions(List<Integer> responseTimes) {
        if (responseTimes.size() == 0 || responseTimes.size() == 1) return 0;
        
        int count = 0;
        long total = responseTimes.get(0);
        double prevAvg = total;
        for (int i = 1; i < responseTimes.size(); i++){
            int curr = responseTimes.get(i);
            
            if (curr > prevAvg){
                count++;
            }
            
            total += curr;
            prevAvg = (double)total / (i + 1);
        }
        return count;

    }

}
